ui/src/createAzureClient.ts

```diff
@@ -56,12 +58,20 @@ export function createAzureSubscriptionClient<T>(clientContext: InternalAzExtCli
     });
 }
 
-export async function sendRequestWithTimeout(context: types.IActionContext, options: RequestPrepareOptions, timeout: number, clientInfo: types.AzExtGenericClientInfo): Promise<HttpOperationResponse> {
+export async function sendRequestWithTimeout(context: types.IActionContext, options: types.AzExtRequestPrepareOptions, timeout: number, clientInfo: types.AzExtGenericClientInfo): Promise<HttpOperationResponse> {
     let request: WebResource = new WebResource();
     request = request.prepare(options);
     request.timeout = timeout;
+
+    if (options.rejectUnauthorized !== undefined) {
+        request.agentSettings = {
+            http: new Agent(),
```

Comment:
What agent does the request default to if you don't set it?  Do we not need to set it even if rejectUnauthorized is undefined?

Reply:
So `agentSettings` is an optional property. However, if you _do_ set it, you have to set both `http` and `https` (I'm not entirely sure why). I assume it just defaults to the standard Agent that I'm using, which is a core part of Node.js